libs/sdk-ui-geo/src/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -202,8 +257,37 @@ function renderPushpinSizeLegend(
     const {
         size: { data, format, name },
     } = geoData;
+    const width = contentRect?.client?.width;
+    const isSmall =
+        !ignoreSmallSize && isAutoPositionWithPopup(responsive) && !shouldRenderWholeCircleLegend(width);
+    const shouldDisplayMeasureName = !ignoreMeasureName && shouldRenderLegendTitles(width);
 
     return (
-        <PushpinSizeLegend format={format} measureName={name} numericSymbols={numericSymbols} sizes={data} />
+        <PushpinSizeLegend
+            format={format}
+            measureName={shouldDisplayMeasureName ? name : undefined}
+            numericSymbols={numericSymbols}
+            sizes={data}
+            isSmall={isSmall}
+        />
+    );
+}
+
+function renderPushpinLegend(
+    props: IGeoChartLegendRendererProps,
+    contentRect: ContentRect,
+    hasSizeLegend: boolean,
+): JSX.Element | null {
+    const { containerId, maxRows, name } = props;
+    return (
+        <PushpinCategoryLegend
+            {...props}
+            contentRect={contentRect}
+            hasSizeLegend={hasSizeLegend}
+            containerId={containerId}
```

Comment:
Nitpick: containerId, maxRows and name are already passed with the `...props` above, right?

Reply:
Fixed https://github.com/gooddata/gooddata-ui-sdk/pull/1341/commits/681c4fcc3655d86e0fca2623bae6c458eb23b47b